src/main/java/org/jenkinsci/plugins/ParameterizedRemoteTrigger/auth2/Auth2.java

```diff
@@ -0,0 +1,85 @@
+package org.jenkinsci.plugins.ParameterizedRemoteTrigger.auth2;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.URLConnection;
+
+import org.jenkinsci.plugins.ParameterizedRemoteTrigger.BuildContext;
+import org.jenkinsci.plugins.ParameterizedRemoteTrigger.exceptions.CredentialsNotFoundException;
+
+import hudson.DescriptorExtensionList;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.model.Item;
+import jenkins.model.Jenkins;
+
+public abstract class Auth2 extends AbstractDescribableImpl<Auth2> implements Serializable {
+
+    private static final long serialVersionUID = -3217381962636283564L;
+
+    private static final DescriptorExtensionList<Auth2, Auth2Descriptor> ALL = DescriptorExtensionList
+        .createDescriptorList(Jenkins.getInstance(), Auth2.class);
+
+    public static DescriptorExtensionList<Auth2, Auth2Descriptor> all()
+    {
+        return ALL;
+    }
+
+    public static abstract class Auth2Descriptor extends Descriptor<Auth2>
+    {
+    }
+
+    /**
+     * Tries to identify a user. Depending on the Auth2 type it might be null.
+     *
+     * @param auth
+     *            authorization to trigger jobs in the remote server
+     * @return the user name
+     * @throws CredentialsNotFoundException 
+     *            if the credentials are not found
+     */
+    public static String identifyUser(Auth2 auth) throws CredentialsNotFoundException
+    {
+        if (auth == null) {
+            return null;
+        }
+        else if (auth instanceof TokenAuth) {
+            return ((TokenAuth) auth).getUserName();
+        }
+        else if (auth instanceof CredentialsAuth) {
+            return ((CredentialsAuth) auth).getUserName(null);
+        }
+        else {
+            return null;
```

Comment:
It is a bad design for extension point IIUC. Generally you should consult with all implementations and get username from their API. If you do not use Jenkins extension points intentionally, there should be logging about unsupported type at least